component/distributed_cache/distributed_cache.go

```diff
@@ -93,11 +132,108 @@ func (distributedCache *DistributedCache) SetNextComponent(nextComponent interna
 	distributedCache.BaseComponent.SetNextComponent(nextComponent)
 }
 
-// Start initializes the DistributedCache component without blocking the pipeline.
+// Start : Pipeline calls this method to start the component functionality
+//
+//	this shall not block the call otherwise pipeline will not start
 func (dc *DistributedCache) Start(ctx context.Context) error {
+
 	log.Trace("DistributedCache::Start : Starting component %s", dc.Name())
+
+	cacheDir := "__CACHE__" + dc.cacheID
+	dc.azstroage = dc.NextComponent()
+	for dc.azstroage != nil && dc.azstroage.Name() != "azstorage" {
+		dc.azstroage = dc.azstroage.NextComponent()
+	}
+
+	dc.strorageCallback = initStorageCallback(
+		dc.NextComponent(),
+		dc.azstroage)
+	dc.clusterManager = clustermanager.NewClusterManager(dc.strorageCallback)
+
+	// Check and create cache directory if needed
+	if err := dc.setupCacheStructure(cacheDir); err != nil {
+		return err
+	}
+	log.Info("DistributedCache::Start : Cache structure setup completed")
+
 	return nil
+}
 
+// setupCacheStructure checks and creates necessary cache directories and metadata.
+// It's doing 4 rest api calls, 3 for directory and 1 for creator file.+1 call to check the creator file
+func (dc *DistributedCache) setupCacheStructure(cacheDir string) error {
+	_, err := dc.azstroage.GetAttr(internal.GetAttrOptions{Name: cacheDir + "/Objects"})
+	if err != nil {
+		if os.IsNotExist(err) || err == syscall.ENOENT {
+			directories := []string{cacheDir, cacheDir + "/Nodes", cacheDir + "/Objects"}
+			for _, dir := range directories {
+				if err := dc.azstroage.CreateDir(internal.CreateDirOptions{Name: dir, ForceDirCreationDisabled: true}); err != nil {
+
+					if !bloberror.HasCode(err, bloberror.BlobAlreadyExists) {
+						return logAndReturnError(fmt.Sprintf("DistributedCache::Start error [failed to create directory %s: %v]", dir, err))
+					}
+				}
+			}
+
+		} else {
+			return logAndReturnError(fmt.Sprintf("DistributedCache::Start error [failed to read creator file: %v]", err))
+		}
+	}
+
+	ipaddr, err := getVmIp()
+	if err != nil {
+		return logAndReturnError(fmt.Sprintf("DistributedCache::Start error [Failed to get VM IP : %v]", err))
+	}
+
+	uuidVal, err := common.GetUUID()
+	if err != nil {
+		return logAndReturnError(fmt.Sprintf("DistributedCache::Start error [Failed to retrieve UUID, error: %v]", err))
+	}
+	rvList := make([]dcache.RawVolume, len(dc.cachePath))
+	for index, path := range dc.cachePath {
+		fsid, err := getBlockDeviceUUId(path)
```

Comment:
if more than one cache dirs have the same uuid, we need to complain and skip all but the first one.
It's possible that user gives more than one cache directory on the same filesystem. We don't want more than one RV per filesystem.